tests/kernel/sched/metairq/src/main.c

```diff
@@ -113,6 +115,8 @@ void coop_thread1(void *p1, void *p2, void *p3)
 	cnt2 = coop_cnt2;
 	zassert_equal(cnt2, LOOP_CNT, "Unexpected cnt2 at end: %d", cnt2);
 
+	printk("thread1 end\n");
```

Comment:
debug code? Do we need this in the test?

Reply:
Not necessarily. As the cooperative thread already printed when it ended, I added these for the other two threads. It also makes the failure case more clear in the logs, I think:
```
Running TESTSUITE suite_preempt_metairq
===================================================================
START - test_preempt_metairq
thread1 take sem
thread2 take sem
metairq start
give sem2
thread2 got sem
thread2 loop iteration 0
thread2 loop iteration 1
thread2 loop iteration 2
thread2 loop iteration 3
thread2 end

    Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/sched/metairq/src/main.c:80: metairq_thread: (coop_cnt2 equal to LOOP_CNT)
thread2 wasn't preempted
 FAIL - test_preempt_metairq in 0.065 seconds
```